### PR TITLE
Add Support for secp256k1 algorithms (AKA ES256K)

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

This replaces #428 to add support for the secp256k1 algorithm. The only changes from that original PR are the ones requested on the review:

* Remove the addition of a deprecated method
* Whitespace fixes

It also removes the README txt file.

For review purposes, see the commits for this PR; the changes to the original PR have been captured as separate commits.

### References

#428 

### Testing

- [X] This change adds test coverage
- [ ] This change has been tested on the latest version of Java or why not

How to create a secp256k1 key to be used with ES256K algorithm:

#### Generate the key

`openssl ecparam -genkey -name secp256k1 -out KEY.pem -text`

- The file is split into 2 parts a curve specification (EC PARAMETERS) and the key value
- This format is not suitable to be read by the PermUtils PEM reader so we have to convert it into PCKS8 format:

`openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in KEY.pem -out ec256k-key.private.pem`

#### The public key is then calculated as follows:

`openssl ec -pubout -in KEY.pem -out ec256k-key-public.pem`

#### How to get DER signature:

```
ECDSAAlgorithm algorithm256 = (ECDSAAlgorithm) Algorithm.ECDSA256((ECPublicKey) readPublicKeyFromFile(PUBLIC_KEY_FILE_256K, "EC"), (ECPrivateKey) readPrivateKeyFromFile(PRIVATE_KEY_FILE_256K, "EC"));
String[] parts = ES256K_JWT.split("\\.");

byte[] derSignature = algorithm256.JOSEToDER(Base64.decodeBase64(parts[2]));
String jwt=parts[0]+"."+parts[1]+"."+Base64.encodeBase64URLSafeString(derSignature);
```

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java [hunk 1]
```diff
@@ -157,7 +158,41 @@ public void shouldNotOverwriteKeyIdIfAddedFromRSAAlgorithms() throws Exception {
         String headerJson = new String(Base64.decodeBase64(parts[0]), StandardCharsets.UTF_8);
         assertThat(headerJson, JsonMatcher.hasEntry("kid", "my-key-id"));
     }
-
+    
+    @Test
+    public void shouldAddKeyIdIfAvailableFromECDSAKAlgorithms() throws Exception {
+        ECPrivateKey privateKey = (ECPrivateKey) PemUtils.readPrivateKeyFromFile(PRIVATE_KEY_FILE_EC_256K
```
Comment (lbalmaceda) [inline lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java:164]: weird formatting?

File: lib/src/test/java/com/auth0/jwt/JWTTest.java [hunk 2]
```diff
@@ -494,7 +498,26 @@ public void shouldCreateAnEmptyECDSA256SignedToken() throws Exception {
                 .build();
         assertThat(verified, is(notNullValue()));
     }
+    
+    @Test
+    public void shouldCreateAnEmptyECDSA256KSignedToken() throws Exception {
+        ECPublicKey publicKey = (ECPublicKey) PemUtils.readPublicKeyFromFile(PUBLIC_KEY_FILE_EC_256K, "EC");
+        ECPrivateKey privateKey = (ECPrivateKey) PemUtils.readPrivateKeyFromFile(PRIVATE_KEY_FILE_EC_256K, "EC");
 
+        String signed = JWT.create().sign(Algorithm.ECDSA256K(publicKey, privateKey) );
```
Comment (lbalmaceda) [inline lib/src/test/java/com/auth0/jwt/JWTTest.java:507]: might be good to run the formatter shortcut `option+command+L` to update all these changes
```suggestion
        String signed = JWT.create().sign(Algorithm.ECDSA256K(publicKey, privateKey));
```

File: lib/src/test/java/com/auth0/jwt/PemUtils.java [hunk 3]
```diff
@@ -11,10 +12,7 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.PrivateKey;
 import java.security.PublicKey;
-import java.security.spec.EncodedKeySpec;
-import java.security.spec.InvalidKeySpecException;
-import java.security.spec.PKCS8EncodedKeySpec;
-import java.security.spec.X509EncodedKeySpec;
+import java.security.spec.*;
```
Comment (lbalmaceda) [inline lib/src/test/java/com/auth0/jwt/PemUtils.java:15]: I prefer the other approach but this is a test so it's fine. However, try running the import organizer shortcut `control+option+O`

**Review by lbalmaceda — APPROVED** [review]

File: lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java (hunk 1 above)
Comment (jimmyjames) [reply lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java]: Pushed a commit to fix running the formatter.

File: lib/src/test/java/com/auth0/jwt/JWTTest.java (hunk 2 above)
Comment (jimmyjames) [reply lib/src/test/java/com/auth0/jwt/JWTTest.java]: Yes, added a commit to do just that.

File: lib/src/test/java/com/auth0/jwt/PemUtils.java (hunk 3 above)
Comment (jimmyjames) [reply lib/src/test/java/com/auth0/jwt/PemUtils.java]: Fixed.

**Review by lbalmaceda — APPROVED** [review]

**Comment by lbalmaceda** [comment]
Good to merge 👍

**Pull request merged**